src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -70,9 +70,10 @@ public String greeting(@RequestParam(name = "name", required = false, defaultVal
         return bookRepository.findByTitle(bookTitle);
     }
 
-    @GetMapping("/publisherGenreYear/{publisher}/{genre}/{year}")
-    public List<Book> findByGenrePublisherYear(@PathVariable String publisher, @PathVariable String genre,
-            @PathVariable String year) {
+    @GetMapping("/publisherGenreYear")
```

Comment:
You can use the root path "/", try to use a less descriptive paths

Reply:
fixed at last commit